CWToolsDocs/HtmlTemplates.fs

```diff
@@ -185,19 +217,53 @@ let rec getTypeBlockDepth (depth : int) ((rule, options): NewRule) =
         inner |> List.map (getTypeBlockDepth (depth + 1)) |> List.max
     | _ -> depth
 
-let typeBlock (enums : EnumDefinition list) ((typeDef : TypeDefinition), ((rule, options): NewRule)) =
-    let _, rules = rule |> (function |NodeRule (l, r) -> l, r)
-    let typeBlockDepth = getTypeBlockDepth 0 (rule, options)
+let extractRulesFromRuleType (ruleType: RuleType) : NewRule list =
+    let rules = ruleType |> (function
+        | NodeRule (_, r) -> r
+        | SubtypeRule(_, _, r) -> r
+        | ValueClauseRule (r) -> r
+        | LeafValueRule(_)
+        | LeafRule (_) -> [] )
+    rules
+
+
+let subTypeBlock (enums: EnumDefinition list) ((subTypeDef : SubTypeDefinition)) =   
+    let typeBlockDepth = subTypeDef.rules |> List.map (getTypeBlockDepth 0) |> List.max
```

Comment:
@tboby I'm not sure about this

Reply:
Yeah, this throws an exception for me sometimes, as `List.max` doesn't work on an empty list

Reply:
Perhaps
```
    let typeBlockDepth =
        match subTypeDef.rules |> List.map (getTypeBlockDepth 0) with
        | [] -> 0
        | x -> List.max x
```